### DISCUSSION TITLE
Unavailiable status after reboot of PC and HA

### DISCUSSION THREAD

**Discussion by Vagelis1971** [taligentx/dscKeybusInterface#116] [dscKeybusInterface Discussion]
Hello people.
im getting an unavailiable status if i reboot the pc or HA . Then to make it work properly i need to arm the system manualy and only after that i can see the status in HA . From this point till the next start up everything works fine .. im on an old power 830 . Maybe you guys have the solution for it ?
It look likes its getting into a hibernating mode and it doent wake up. Also in conf.yalm the avaliability status reguest is there.
Thanks in adv

**Comment by taligentx** [comment]
Hi @Vagelis1971 - are you using the master or develop branch?

**Comment by Vagelis1971** [comment]
Ahhh. found it .. im using the master...

**Comment by Vagelis1971** [comment]
I have upgraded in 1.2 version and the same thing. Im still having that problem i mentioned before . im on a nodemcu V1 . Actualy had the same problem with tamota sonoff mqtt software. but that i sovled by  adding an automation that restarts all my nodemcu tasmota software as HA starts with a delay of 20 sec. All the unavailiable entities got back each time win10 auto restarts HA after an electrical cut out . I tried to reset it manually from nodemcu board but same thing.

**Comment by taligentx** [comment]
The develop branch should handle this, it's one of the changes since the last release.

**Comment by Vagelis1971** [comment]
k ill try it tonight and let you know. thanks

**Comment by taligentx** [comment]
To troubleshoot this, try subscribing to the `dsc/Status` MQTT topic on a different computer and see which messages show up while you reboot the HA and MQTT system.  Do you have HA and the MQTT broker on the same system?

The develop branch sketch tracks the connection to the MQTT broker and sends a full status update whenever the MQTT broker connection is reconnected, the sketch doesn't check for HA status.  HA should be picking up the status from the MQTT broker because the MQTT messages from the sketch are marked as persistent, even if HA comes online after the status update is sent, HA should receive the current status from the MQTT broker once HA is online.

**Comment by taligentx** [comment]
Closing for lack of updates, feel free to re-open if this is still an issue.
